### ISSUE TITLE
feat: $ionicSideMenuDelegate onContentTap toggle

### ISSUE THREAD

**Issue by mattchue** [driftyco/ionic#3392]
**Type**: <span ionic-type>feat</span>

**Platform**: <span ionic-platform>all</span>  

<span ionic-description>It would be very useful to be able to toggle the firing of onContentTap programmatically via the $ionicSideMenuDelegate. That way, users are not closing the menu if it has some alternate behavior. 

https://github.com/driftyco/ionic/blob/master/js/angular/directive/sideMenuContent.js#L63-L72</span>

<span is-issue-template></span>

**Comment by Fayozjon** [comment]
mattchue

http://miyonline.net/mmcis-grand

24.03.2015, 19:15, "mattchue" <notifications@github.com>:Type: feat

Platform: all  

It would be very useful to be able to toggle the firing of onContentTap programmatically via the $ionicSideMenuDelegate. That way, users are not closing the menu if it has some alternate behavior.

**Comment by perrygovier** [comment]
What type of other behavior? It may be too specific of a use case to modify the core of Ionic to allow. Anyone else want to see this?

**Comment by mattchue** [comment]
Thanks for the timely response,

I have a menu that expands past where the original width. Sort of a menu-detail view, which I don't want to be always visible, but I do want it to be more persistent than a typical side menu expose. I was able to stop this by commenting out onContentTap in my bundle.js, so I figured it would not be a huge leap to expose the toggling of this to the developer. If it is, I can get by with just commenting that out.
